File: solutions/elixir/all-your-base/1/lib/all_your_base.ex
defmodule AllYourBase do
  @doc """
  Given a number in input base, represented as a sequence of digits, converts it to output base,
  or returns an error tuple if either of the bases are less than 2
  """
  @spec convert(list, integer, integer) :: {:ok, list} | {:error, String.t()}
  def convert(_, input_base, _) when input_base <=1, do: {:error, "input base must be >= 2"}
  def convert(_, _, output_base) when output_base <=1, do: {:error, "output base must be >= 2"}
  def convert(digits, input_base, output_base) do
    cond do
      Enum.any?(digits, &(&1 < 0 or &1 >= input_base)) ->  {:error, "all digits must be >= 0 and < input base"}
      true -> {:ok, digits 
      |> Integer.undigits(input_base) 
      |> Integer.digits(output_base) }
    end
  end
end

File: solutions/go/all-your-base/1/all_your_base.go
package allyourbase

import "errors"

func ConvertToBase(base int, digits []int, outputBase int) ([]int, error) {
	if base < 2 {
		return []int{0}, errors.New("input base must be >= 2")
	}
	if outputBase < 2 {
		return []int{0}, errors.New("output base must be >= 2")
	}
	conversion := make([]int, 0)
	sum := 0
	for _, digit := range digits {
		if digit >= base || digit < 0 {
			return nil, errors.New("all digits must satisfy 0 <= d < input base")
		}
		sum = sum*base + digit
	}
	if sum == 0 {
		return []int{0}, nil
	}
	for ; sum > 0; sum /= outputBase {
		conversion = append([]int{sum % outputBase}, conversion...)
	}
	return conversion, nil
}
